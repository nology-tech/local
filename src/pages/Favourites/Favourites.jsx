import Card from "../../components/Card/Card";
import Layout from "../../components/Layout/Layout";
import DropDown from "../../components/DropDown/DropDown";
import Filter from "../../components/Filter/Filter";
import { useContext, useState } from "react";
import UserContext from "../../context/UserContext";
import { handleUserFavouritesUpdate } from "../../api/userService";

const Favourites = () => {
  const { user } = useContext(UserContext);
  if (user === null) {
    return <p>loading...</p>;
  }

  const [allCardsArr, setAllCardsArr] = useState(user.favourites);

  const optionsArr = [
    "All",
    "London",
    "Birmingham",
    "Chester",
    "Oxford",
    "Newcastle",
    "Liverpool",
  ];

  const handleRemoveFavourites = (cardId) => {
    let localUserFavouritesArray = user.favourites;

    const updatedUserFavourites = localUserFavouritesArray.filter(
      (element) => element._id !== cardId
    );
    handleUserFavouritesUpdate(updatedUserFavourites, user.id);
  };

  const onChange = (location) => {
    console.log(location);
  };

  return (
    <Layout isWithMenu={true}>
      <DropDown onChange={onChange} options={optionsArr} />
      <Filter favArray={user.favourites} setAllCardsArr={setAllCardsArr} />
      <Card
        cardArray={allCardsArr}
        handleRemoveFavourites={handleRemoveFavourites}
      />
    </Layout>
  );
};
export default Favourites;
