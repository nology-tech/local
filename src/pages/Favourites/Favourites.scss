@use "../../styles/variables" as *;

.favourites__container {
  display: flex;
  flex-flow: column nowrap;
  align-items: stretch;
  padding: 0 1rem;
}
