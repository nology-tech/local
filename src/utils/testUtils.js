import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
