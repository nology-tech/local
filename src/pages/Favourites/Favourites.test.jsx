import { customRender } from "../../utils/testUtils";
import Favourites from "./Favourites";

it("should render the text on the display", () => {
  const { container } = customRender(<Favourites />);

  expect(container).toMatchSnapshot();
});
