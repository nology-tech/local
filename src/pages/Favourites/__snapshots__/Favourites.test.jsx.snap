// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="layout__header"
    >
      <div
        class="header"
      >
        <img
          alt=""
          src="BackButton.svg"
        />
        <div
          class="header__container"
        >
          <div
            class="header__logo-title"
          >
            <img
              alt=""
              src="AppLogo.svg"
            />
            <h1
              class="header__title-local"
            >
              LOCAL
            </h1>
          </div>
        </div>
      </div>
      <img
        alt="Menu icon"
        src="menu.svg"
      />
    </div>
    <div
      class="dropdown"
    >
      <div
        class="dropdown__item dropdown__item--primary"
        role="select"
        tabindex="0"
      >
        <p
          class="dropdown__item-name dropdown__item-name--primary"
        >
          All
        </p>
        <div
          class="dropdown__image-container"
        >
          <img
            alt="triangle"
            class="dropdown__arrow"
            src="triangleUpsideDown.svg"
          />
        </div>
      </div>
      <div
        class="dropdown__options"
      />
    </div>
    <div>
      <h2 />
      <p>
        Medicine Bakery is a small, family-run bakery located in the city of Birmingham. The bakery specializes in creating delicious, wholesome, and nutritious baked goods, all of which are made with only the freshest ingredients. From savory pies and tarts to sweet treats like muffins and cupcakes, Medicine Bakery has something to delight every palate. The bakery also offers vegan and gluten-free options, making it the perfect spot to enjoy a guilt-free snack or treat. Stop by and try something from Medicine Bakery today!
      </p>
      <p>
        69a New St, Birmingham B2 4DU
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
    <div>
      <h2 />
      <p>
        A ladies' clothes shop offering a wide range of fashionable pieces.
      </p>
      <p>
        5 N Western arcade, Birmingham B2 5LH
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
    <div>
      <h2 />
      <p>
        A restaurant and Indian restaurant known for its delicious food and authentic flavors.
      </p>
      <p>
        21a Bennetts Hill, Birmingham B2 5QP
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
    <div>
      <h2 />
      <p>
        A pub, bar, and more, offering a wide range of drinks and a fun atmosphere.
      </p>
      <p>
        8 Bennetts Hill, Birmingham B2 5RS
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
    <div>
      <h2 />
      <p>
        This is a lively bar and restaurant that serves a variety of beers, wines, and cocktails, as well as classic pub food like burgers, sandwiches, and wings. It has a fun, casual atmosphere and is a popular spot for watching sports.
      </p>
      <p>
        79 Seel St, Liverpool L1 4BB
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
    <div>
      <h2 />
      <p>
        A fun and quirky shop featuring a range of products and gifts inspired by local culture and humor.
      </p>
      <p>
        31-33 Tarleton St, Liverpool L1 1DS
      </p>
      <div>
        <button
          class="button button--details"
        >
          Details
        </button>
        <button
          class="button button--remove"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
